src/qibocal/protocols/characterization/signal_experiments/tof_readout.py

```diff
@@ -0,0 +1,131 @@
+from dataclasses import dataclass, field
+from typing import Dict, Optional, Union
+
+import numpy as np
+import plotly.graph_objects as go
+from plotly.subplots import make_subplots
+from qibolab import AcquisitionType, AveragingMode, ExecutionParameters
+from qibolab.platform import Platform
+from qibolab.pulses import PulseSequence
+
+from qibocal.auto.operation import Parameters, Qubits, Results, Routine
+from qibocal.data import DataUnits
+
+
+@dataclass
+class ToFParameters(Parameters):
+    """ToF runcard inputs."""
+
+    nshots: Optional[int] = None
+    """Number of shots."""
+    relaxation_time: Optional[int] = None
+    """Relaxation time (ns)."""
+
+
+@dataclass
+class ToFResults(Results):
+    """ToF outputs."""
+
+    ToF: Dict[Union[str, int], float] = field(metadata=dict(update="time_of_flight"))
```

Comment:
```suggestion
    ToF: Dict[QubitId, float] = field(metadata=dict(update="time_of_flight"))
```
You can do this by importing `QubitId` from `qibolab.qubits`